### PR TITLE
Big repository restructuring

### PR DESCRIPTION
**Pull request by CFSworks**
I realize this PR is the equivalent of a sledgehammer to a load-bearing wall: it's a pretty major change, but I think it (or something very much like it) is a necessary one, for two reasons:

1. The current build process involves too many manual copies of config files. This is developer-unfriendly and error-prone, and I strongly believe we need to do something about this sooner rather than later.
2. The way the repository is structured now, it isn't clear what is modified for the TP2 itself vs. upstream components that can be upgraded in the future. Commingling our own edits with the upstream components (namely, buildroot) tends to "ossify" our local copies, making them harder and harder to update in the future. It's best to nip that in the bud and cleanly separate our modifications *now.* Having our stuff cleanly separated out also helps new developers get up-to-speed quickly.

I have made sure that everything which I have moved is recognized as a "rename" by Git, so that merges made from branches forked before this reorganization will still apply relatively cleanly. There will be a little bit of work downstream of this, but that inevitable pain will only compound the more we let this current structure solidify.

This PR should also *probably* be applied to the legacy wenyi0421 repo first, and later merged down here from there, as I don't want the legacy and community repositories to drift too far away from this one. For that reason, I'm opening this PR here as a draft so that the changes can be discussed with the goal of getting approval of Turing Machines itself before I try to introduce this PR over there.

I welcome any feedback (even the very nitpicky stuff!) and am more than happy to apply most adjustments - my only real goal is to solve the 2 concerns I have above.

### CODE REVIEW

**Comment by svenrademakers** [comment]
The work you did was due for a long time, thanks. I quickly browsed through it and could not see big blockers.
I do need to do some manual verification on it, So please give me a bit of time to go over this.

Keeping in sync with wenyi042 is up to you. I hope in the nearby future we can consolidate the "community" editions with this repo. As it is our mission to serve the community. Expect more activity on this repo soon

**Review by svenrademakers — APPROVED** [review]
i verified the PR.  Flashed it to the board, even though its only restructuring of the repo.
I would like to submit this if your ok with it

File: readme.md [hunk 1]
```diff
@@ -6,34 +6,18 @@ sudo apt-get install build-essential subversion git-core libncurses5-dev zlib1g-
...
-//build swu
+# Package up swu (for OTA update via webui):
 cd output/images/
-./genSWU.sh 1.0.0	//build swu
+cp -ra ../../../tp2bmc/swupdate/* .
+./genSWU.sh 1.0.0
 
-//Execute mkfw.sh 1.0.0 in the root directory to directly compile the firmware to the build directory
+# ...or, execute this in the main directory to package both .img and .swu:
 ./mkfw.sh 1.0.0
```
Comment (svenrademakers) [inline readme.md:22]: this script actually doesn't work when you run it on a clean repo.
as the build targets are not yet configured in make. which line 12 does.
Could you make a comment about this in the meantime? as mentioned in my other comment, i will do a cleanup pass shortly

Comment (CFSworks) [reply readme.md]: I decided resolve this by instead doing away with this script entirely.

File: readme.md [hunk 2]
```diff
@@ -6,34 +6,18 @@ sudo apt-get install build-essential subversion git-core libncurses5-dev zlib1g-
...
-//update config  //Only once
-cd ../
-cp bmc4tpi/config/sun8iw20p1* buildroot/output/build/linux-5112fdd843715f1615703ca5ce2a06c1abe5f9ee/arch/arm/boot/dts/
-cp bmc4tpi/config/kernelconfig buildroot/output/build/linux-5112fdd843715f1615703ca5ce2a06c1abe5f9ee/.config
-cp bmc4tpi/config/swupdateconfig buildroot/output/build/swupdate-2021.11/.config
-cp bmc4tpi/swupdate/sw-description buildroot/output/images/
-cp bmc4tpi/swupdate/genSWU.sh buildroot/output/images/
-cp bmc4tpi/swupdate/env0.fex buildroot/output/images/
-cp bmc4tpi/swupdate/env1.fex buildroot/output/images/
+make   BR2_EXTERNAL=../tp2bmc  tp2bmc_defconfig
```
[the remark this answers is not in the document]
Comment (CFSworks) [reply readme.md:12]: Whatever we do decide, we should also set up GitHub Actions to run it. It would save a step reviewing PRs, too.

**Comment by CFSworks** [comment]
>I would like to submit this if your ok with it

I'll flip this from draft to a "ready" PR then!

**Review by svenrademakers — APPROVED** [review]

**Comment by svenrademakers** [comment]
This PR was merged successfully. I only wrongly rebased it, so i had to manually close this PR

**Comment by ruslashev** [comment]
Thank you! Also thanks for .gitignore changes, and getting rid of config copying via `BR2_PACKAGE_SWUPDATE_CONFIG` and others.

**Pull request closed without merging**